### ISSUE TITLE
spurious warning (W0640) issued for loop variables used in local functions

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#271]
Originally reported by: **BitBucket: [hayne](http://bitbucket.org/hayne), GitHub: @hayne?**

---

The following code that has a local function defined inside a loop triggers a W0640 warning ("Cell variable defined in loop") in PyLint 1.2.1

```
#!python

for i in range(10):
    def printLoopVar():
        print i
    printLoopVar()
```

This warning seems to be spurious since there is no problem with the above code.
The variable 'i' which is used in the local function 'printLoopVar' is NOT part of a closure as I understand them. Moreover, this code works perfectly well and gives the expected results.

I think the warning is intended to catch cases like the following where there is a real problem - however PyLint 1.2.1 does not issue any warning here.

```
#!python

funcList = list()
for i in range(10):
    def func():
        return i
    funcList.append(func)
for f in funcList:
    print f()
```

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/271
